### ISSUE TITLE
True loss values

### ISSUE THREAD

**Issue by sparklingdew** [tensorflow/tensorflow#47370]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 1.15.3 and 2.4.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
The log loss values of the training dataset recorded during model fitting are not the actual loss values obtained when evaluating the resulting model. I explain this througly in https://github.com/sparklingdew/keras_loss_discrepancy.

**Will this change the current api? How?**
The only change that I suggest is to give the actual loss value for the training dataset, rather than this other internal value which is difficult to understand. It should be calculated similarly to the validation loss, which is obtained from evaluating the resulting model. I understand that this may go in detriment to running time. In that case, this new feature could be optional.

**Who will benefit with this feature?**
NN users will be benefitted, since model tunning is performed based on loss and accuaracy results. If these values are misleading, the models are difficult to tune.

**Any Other info.**
Many thanks for your time!

**Comment by jvishnuvardhan** [comment]
@sparklingdew Sorry for the late response. Are you still interested in contributing? If yes, please feel free to open a PR in  [keras-team/keras repo.](https://github.com/keras-team/keras/issues) repository.

Please note that Keras development moved to keras-team/keras repository to focus on only keras. Thanks!
